plugins/vbe/fu-vbe-plugin.c

```diff
@@ -79,33 +77,41 @@ fu_vbe_plugin_coldplug_img(FuPlugin *plugin,
 	/* success */
 	dev = g_object_new(device_gtype,
 			   "context",
-			   fu_plugin_get_context(plugin),
+			   fu_plugin_get_context(FU_PLUGIN(self)),
 			   "fdt-root",
 			   fdt_root,
 			   "fdt-node",
 			   fdt_node,
 			   NULL);
-	fu_plugin_device_add(plugin, dev);
+	fu_plugin_device_add(FU_PLUGIN(self), dev);
 	return TRUE;
 }
 
 static gboolean
-fu_vbe_plugin_coldplug(FuPlugin *plugin, FuProgress *progress, GError **error)
+fu_vbe_plugin_backend_device_added(FuPlugin *plugin, FuDevice *device, GError **error)
 {
 	FuVbePlugin *self = FU_VBE_PLUGIN(plugin);
+	g_autoptr(FuFirmware) fdt = NULL;
 	g_autoptr(FuFdtImage) fdt_root = NULL;
+	g_autoptr(FuProgress) progress = fu_progress_new(G_STRLOC);
 	g_autoptr(GPtrArray) fdt_imgs = NULL;
 
+	/* not interesting */
+	if (g_strcmp0(fu_device_get_backend_id(device), "fdt") != 0)
+		return TRUE;
+
 	/* get compatible from root node */
-	fdt_root =
-	    fu_fdt_firmware_get_image_by_path(FU_FDT_FIRMWARE(self->fdt), "/chosen/fwupd", error);
+	fdt = fu_device_read_firmware(device, progress, error);
```

Comment:
this is a hack to avoid moving the FuFdtDevice to libfwupdplugin and having a `fu_fdt_device_get_firmware()` export -- if you think that's preferable let me know and I can move it.

Reply:
Ooooor, we could just have a fu_context_get_fdt() and return the FuFdtFirmware there; loading it with all the other hwinfo context stuff. The disadvantage with that if you can't do instance ID quirk matches but maybe that's not the end of the world.

Reply:
I like https://github.com/fwupd/fwupd/pull/5407 more